plugin.video.composite_for_plex/resources/lib/composite_addon/plex/plex.py

```diff
@@ -0,0 +1,797 @@
+        }
+
+        self.setup_user_token()
+        if load:
+            self.load()
+
+    def plex_identification_header(self):
+        self.client_id = get_client_identifier(self.settings, self.client_id)
+        return create_plex_identification(self.settings, client_id=self.client_id,
+                                          token=self.effective_token)
+
+    def is_plexhome_enabled(self):
+        return self.plexhome_settings['plexhome_enabled']
+
+    def is_myplex_signedin(self):
+        return self.plexhome_settings['myplex_signedin']
+
+    def get_myplex_user(self):
+        return self.effective_user
+
+    def get_myplex_avatar(self):
+        return self.plexhome_settings['plexhome_user_avatar']
+
+    def signout(self):
+        self.plexhome_settings = {
+            'myplex_signedin': False,
+            'plexhome_enabled': False,
+            'myplex_user_cache': '',
+            'plexhome_user_cache': '',
+            'plexhome_user_avatar': ''
+        }
+
+        self.delete_cache(True)
+        LOG.debug('Signed out from myPlex')
+
+    def get_signin_pin(self):
+        data = self.talk_to_myplex('/pins.xml', method='post')
+        try:
+            xml = ETree.fromstring(data)
+            code = xml.find('code').text
+            identifier = xml.find('id').text
+        except:  # pylint: disable=bare-except
+            code = None
+            identifier = None
+
+        if code is None:
+            LOG.debug('Error, no code provided')
+            code = '----'
+            identifier = 'error'
+
+        LOG.debug('code is: %s' % code)
+        LOG.debug('id   is: %s' % identifier)
+
+        return {
+            'id': identifier,
+            'code': list(code)
+        }
+
+    def check_signin_status(self, identifier):
+        data = self.talk_to_myplex('/pins/%s.xml' % identifier, method='get2')
+        xml = ETree.fromstring(data)
+        temp_token = xml.find('auth_token').text
+
+        LOG.debugplus('Temp token is: %s' % temp_token)
+
+        if temp_token:
+            response = requests.get('%s/users/account?X-Plex-Token=%s' %
+                                    (self.myplex_server, temp_token),
+                                    headers=self.plex_identification_header())
+            response.encoding = 'utf-8'
+
+            LOG.debug('Status Code: %s' % response.status_code)
+
+            if response.status_code == 200:
+                try:
+                    LOG.debugplus(encode_utf8(response.text, py2_only=False))
+                    LOG.debug('Received new plex token')
+                    xml = ETree.fromstring(encode_utf8(response.text))
+                    home = xml.get('home', '0')
+                    username = xml.get('username', '')
+
+                    avatar = xml.get('thumb')
+                    # Required because plex.tv doesn't return content-length and
+                    # KODI requires it for cache
+                    # fixed in KODI 15 (isengard)
```

Comment:
not needed anymore (you're submiting to leia)

Reply:
was an old invalid comment, removed